src/hooks/sarcoToken/useSarcoBalance.ts

```diff
@@ -0,0 +1,32 @@
+import { useAccount, useContractRead } from 'wagmi';
+import { SarcoTokenMock__factory } from '@sarcophagus-org/sarcophagus-v2-contracts';
+import { useNetworkConfig } from 'lib/config';
+import { BigNumber, ethers } from 'ethers';
+import { useMemo } from 'react';
+
+export function useSarcoBalance() {
+  const { address, isConnected } = useAccount();
+  const networkConfig = useNetworkConfig();
+  const { data, isError, isLoading } = useContractRead({
+    address: networkConfig.sarcoTokenAddress,
+    abi: SarcoTokenMock__factory.abi,
+    functionName: 'balanceOf',
+    args: [address],
+  });
```

Comment:
what if we added 
`enabled: isConnected,`
to the `useContractRead`, then we could remove it as a dependency to 'formattedBalance'? Is that a good idea?

Reply:
Actually I reconsidered my 1st comment. I think we can remove `isConnected` all together.
and add
`enabled: !!address` to `useContractRead()`

my final thought on this is, maybe we don't need to worry about it at all, and than wagmi will take care of all this for us?

Reply:
Interesting. Thanks, thanks